### PR TITLE
[*] FO : Added title tags to links and images

### PR DESCRIPTION
**Pull request by Dh42**
Title is expanitory

### CODE REVIEW

**Comment by djfm** [comment]
Hi, I see no objection to merging this, but can you explain a bit more why this is useful please?

According to [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title) the `title` attribute is mostly used on `link`, media elements, `p`, and quotations.

The `img` already has an `alt` and the link already has a content (its body). Does it really bring value to add a `title`?

**Comment by Dh42** [comment]
It is purely for SEO, it is considered an SEO best practice. The title tag is also used by screen readers where the alt is not. Also the title tag is used for tool tips as well.

**Comment by inetbiz** [comment]
@djfm http://www.w3.org/html/wg/drafts/html/master/dom.html#the-title-attribute I'm not so certain screen readers use the title tag after reading that document. SEO is not in best practices for images. The alt tag is used for image indexing as you suggested. :-1: this PR.

**Comment by Dh42** [comment]
Every major SEO crawler will pick it up as an error. Moz, Raven, Yoast recommends it.

https://yoast.com/image-seo-alt-tag-and-title-tag-optimization/

And SEO is great for images too. We have a lot of clients that pull a lot of traffic from image searches. 

It seems like a no brainer to add, can you find one source that discourages it?

**Comment by inetbiz** [comment]
@Dh42 I had merely read the specs, first. But then, https://support.google.com/webmasters/answer/114016?hl=en talks about image titles and placing images near contextual information. One interesting idea would be using <code>< caption >~</ caption ></code>

**Comment by Dh42** [comment]
A caption tag is a visible element, you don't want to use visible elements, they break the design layout and are generally messy. Think of a caption tag like a head tag, or a menu tag. It is block level. Plus it is only recommended for table use, https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption and http://www.w3.org/TR/html-markup/caption.html

**Comment by inetbiz** [comment]
@Dh42 wrong selection. http://www.w3schools.com/Tags/tag_figcaption.asp

File: themes/StarterTheme/templates/catalog/product-miniature.tpl [hunk 1]
```diff
@@ -5,13 +5,14 @@
       <img
         src = "{$product.cover.medium.url}"
         alt = "{$product.cover.legend}"
+        title = "{$product.cover.legend}"
```
Comment (inetbiz) [inline themes/StarterTheme/templates/catalog/product-miniature.tpl]: Title tag has a limitation in characters. It should not duplicate the alt tag content. Perhaps modifying the Back Office and a new record field.

**Comment by Dh42** [comment]
Its considered good SEO practice to use it this way. If more features are added, such as a field for this in the BO, the back office starts becoming over complicated.

**Comment by inetbiz** [comment]
Title tag is good as I have learned. But, duplicating field content is **not** best practice.

**Comment by Dh42** [comment]
Its not, correct, but having to granular control will in the end slow things down and add too much of a layer of complication.

**Comment by inetbiz** [comment]
There is no wisdom in adding a title tag just for the sake of having it if the content of that title is not unique or appropriate in its use.

**Comment by Dh42** [comment]
Sure there is. It still works in ranking a site for a keyword.

**Comment by inetbiz** [comment]
Not really. I pointed out from the W3 draft that the title field is ignored because it is not useful on certain devices such as keyboard only and touch devices. Search is pushing mobile. A title field is not even specified for the  tag. The title came from global attributes. Unless we have unique titles, I would not recommend this PR at this time.

**Comment by Dh42** [comment]
The W3C is a consortium on writing standardized code, they have nothing to do with what search engines consider a ranking factor. To be quite honest Google does not even use their standards. Valid code is not and has never been a ranking factor, but keyword density has always been.

**Comment by inetbiz** [comment]
Keyword density measured from onpage content. I did mention switching to forward facing content for semantics and keywords.

```
<figure>
  <img src="img_pulpit.jpg" alt="The Pulpit Rock" width="304" height="228">
  <figcaption>Fig1. - A view of the pulpit rock in Norway.</figcaption>
</figure>
```

**Comment by inetbiz** [comment]
I'm well aware of what the W3C does and how browsers and search engine giant's work together on those drafts. Show me a single instruction that tells you to duplicate your alt tag for the title tag?

**Comment by Dh42** [comment]
This one PR is becoming a time suck for me and I am going to abandon it. 

You are trying to push your wants with limited understanding on a community and I am trying to push long accepted best practices. I don't know if you realize it or not, but the snippet you keep posting is adding a block level element. This is what it looks like in practice, http://www.w3schools.com/Tags/tryit.asp?filename=tryhtml5_figcaption  Do you really want an on page caption element on every image of a site?  This is unneeded and looks ugly.

As for keyword density it is calculated in many ways. PrestaShop site don't just target Google. They target every major and minor search engine in the world. Search engines such as Yandex still use a different algorithm to calculate KWD.  

I am going to leave this at I am suggesting what every major SEO person in the industry is in agreement with. You are suggesting things of wants from your end.

**Comment by kpodemski** [comment]
:+1: for this pr

**Comment by inetbiz** [comment]
Can't we debate code without resorting to personal attacks?

**Comment by Dh42** [comment]
What was the personal attack?

**Comment by inetbiz** [comment]
I won't drag out the insults on this PR. You're a wonderful module developer. I just do not feel... strongly do not feel that duplicating the alt tag is beneficial in the slightest and I'm not the only one saying this in social media. POSH code is beneficial to machine learning. The alt and title tag MUST stay unique. Greyhat SEO is just too risky.

**Comment by web-seo** [comment]
Adding title tags to images will have no positives for seo. Not only for Prestashop, for any other CMS. There is no sense to use it only as duplicate for alt tags.

**Comment by web-seo** [comment]
P.S. Titles for images and for links is one of over 300 myths for Google ranking factors. More about here: https://northcutt.com/wr/google-ranking-factors/  - filter by myth and search for "Keyword Link Title".﻿ On source link you will find the same for image titles. LIke this: https://www.searchenginejournal.com/image-alt-text-vs-image-title-whats-the-difference/ - article says duplicate alt tags as title tags is sligtly negative factor.﻿

**Pull request closed without merging**